crates/bevy_pbr/src/environment_map/generate_from_skybox.rs

```diff
@@ -0,0 +1,689 @@
+///
+/// Usage:
+/// * Add this component via `GenerateEnvironmentMapLight::default()` to an entity with a [`Skybox`] component.
+/// * The first frame this component is added to the skybox entity, an [`EnvironmentMapLight`]
+/// component will be generated and added to the skybox entity.
+/// * For static (non-changing) skyboxes, remove this component 1 frame after adding it to the skybox entity to save performance.
+///
+/// This component does not work on `WebGL2`, and must use [`GenerateEnvironmentMapLightTextureFormat::Rgba16Float`] on `WebGPU`.
+#[derive(Component, ExtractComponent, Reflect, Default, Clone)]
+pub struct GenerateEnvironmentMapLight {
+    texture_format: GenerateEnvironmentMapLightTextureFormat,
+    downsampled_cubemap: Option<Handle<Image>>,
+}
+
+impl GenerateEnvironmentMapLight {
+    pub fn new_with_texture_format(
+        texture_format: GenerateEnvironmentMapLightTextureFormat,
+    ) -> Self {
+        Self {
+            texture_format,
+            downsampled_cubemap: None,
+        }
+    }
+}
+
+#[derive(Reflect, Clone, Copy)]
+pub enum GenerateEnvironmentMapLightTextureFormat {
+    /// 4 bytes per pixel (smaller and faster), but may not be able to represent as wide a range of lighting values.
+    /// This is the [`Default`] on non-WASM platforms.
+    Rg11b10Float,
+    /// 8 bytes per pixel. This is the [`Default`], and only supported option for `WebGPU`.
+    Rgba16Float,
+}
+
+impl Default for GenerateEnvironmentMapLightTextureFormat {
+    fn default() -> Self {
+        if cfg!(target_arch = "wasm32") {
+            Self::Rgba16Float
+        } else {
+            Self::Rg11b10Float
+        }
+    }
+}
+
+impl GenerateEnvironmentMapLightTextureFormat {
+    pub fn as_wgpu(&self) -> TextureFormat {
+        match self {
+            Self::Rg11b10Float => TextureFormat::Rg11b10Float,
+            Self::Rgba16Float => TextureFormat::Rgba16Float,
+        }
+    }
+}
+
+#[derive(Resource)]
+pub struct GenerateEnvironmentMapLightResources {
+    rg11b10float: GenerateEnvironmentMapLightResourcesSpecialized,
+    rgba16float: GenerateEnvironmentMapLightResourcesSpecialized,
+    filter_coefficents: Buffer,
+}
+
+impl FromWorld for GenerateEnvironmentMapLightResources {
+    fn from_world(world: &mut World) -> Self {
+        let render_device = world.resource::<RenderDevice>();
+        let pipeline_cache = world.resource::<PipelineCache>();
+
+        Self {
+            rg11b10float: GenerateEnvironmentMapLightResourcesSpecialized::new(
+                TextureFormat::Rg11b10Float,
+                render_device,
+                pipeline_cache,
+            ),
+            rgba16float: GenerateEnvironmentMapLightResourcesSpecialized::new(
+                TextureFormat::Rgba16Float,
+                render_device,
+                pipeline_cache,
+            ),
+            filter_coefficents: render_device.create_buffer_with_data(&BufferInitDescriptor {
```

Comment:
We're wasting ~40kb of GPU memory on this if the user doesn't actually need this functionality. Perhaps this plugin should not be on by default.

Reply:
Gonna be honest, 40kb of of VRAM is nothing these days.

Still probably worth not having it on by default (not sure how big of a change that is though).

Reply:
It would probably be making the plugin not part of DefaultPlugins, or putting it in a OnceCell initialized on first access or something.